app/lib/common/providers/room_providers.dart

```diff
@@ -169,32 +169,41 @@ final spaceRelationsProvider = FutureProvider.autoDispose
   return await room.spaceRelations();
 });
 
-/// Get the canonical parent of the space. Errors if the space isn't found. Stays up
+/// Get the canonical parents of the space. Errors if the space isn't found. Stays up
 /// to date with underlying client data if a space was found.
-final canonicalParentProvider = FutureProvider.autoDispose
-    .family<SpaceWithProfileData?, String>((ref, roomId) async {
+final canonicalParentsProvider = FutureProvider.autoDispose
+    .family<List<SpaceWithProfileData>, String>((ref, roomId) async {
   try {
+    List<SpaceWithProfileData> parentList = List.empty(growable: true);
     final relations = await ref.watch(spaceRelationsProvider(roomId).future);
     if (relations == null) {
-      return null;
-    }
-    final parent = relations.mainParent();
-    if (parent == null) {
-      return null;
+      return parentList;
     }
 
-    final parentSpace =
-        await ref.watch(maybeSpaceProvider(parent.roomId().toString()).future);
-    if (parentSpace == null) {
-      return null;
+    // Collect all parents: mainParent and otherParents
+    final allParents = [];
+    final mainParent = relations.mainParent();
+    if (mainParent != null) {
+      allParents.add(mainParent);
     }
-    final profile =
-        await ref.watch(spaceProfileDataProvider(parentSpace).future);
-    final SpaceWithProfileData data = (space: parentSpace, profile: profile);
-    return data;
+    allParents.addAll(relations.otherParents());
+
+    for (final parent in allParents) {
+      final parentSpace = await ref
+          .watch(maybeSpaceProvider(parent.roomId().toString()).future);
+      if (parentSpace == null) {
+        continue;
+      }
+      final profile =
+          await ref.watch(spaceProfileDataProvider(parentSpace).future);
+      final SpaceWithProfileData data = (space: parentSpace, profile: profile);
```

Comment:
this isn't really safe for two reasons: if we don't know the space itself, you are just dropping it -- we haven't really agreed this is the right thing to do in general but at the very least it should at least say this here with a comment to make clear we are making this assumption of just skipping it.

secondly, if any _one_ parent fails to load the avatar or info, the entire list won't show and all we have to go for figuring this out is a warning in the logs. Even if a profile picture fails to load we should show the fallback space icon and log the error.

I think it'd be best if we refactored the providers to directly generate the AvatarInfo and update it upon successful loading of the data rather than silently dropping it.

Reply:
I'll see how it can be changed regarding the proposal you've suggested. I agree it's better to safely load list regardless of internal data availability of item.